Update licenses, junit version

vinayvishal opened the pull request.

bshannon reviewed (changes requested):
The Junit version update is fine.
The spec license should not be removed.
The copyright text was purposely left off the mailcap and mime.types files to avoid the runtime overhead of reading and discarding all those lines.  Please don't change those files.

vinayvishal commented:
1. Reverted the deletion of speclicense.html.

1. Reverted the changes done in mime.types and mailcap files.

1. Deleted all but *.txt files from doc/spec as per the review comments.

1. LICENSE.txt will be updated in corresponding EF repository.

bshannon reviewed (changes requested):
Comments sent by email.

Commit:
Update Junit version
